cypress/e2e/gui/customers.cy.js

```diff
@@ -0,0 +1,299 @@
+      cy.contains('button', 'View').click();
+      cy.contains('button', 'Back').click();
+
+      cy.contains('p', 'Below is our customer list.').should('be.visible');
+    });
+
+    it('shows a Loading... fallback element before the initial customers fetch', () => {
+      cy.intercept(
+        'GET',
+        `${CUSTOMERS_URL}?page=1&limit=10&size=All&industry=All`,
+        (req) => {
+          req.reply((res) => {
+            res.setDelay(4000);
+          });
+        }
+      ).as('getDelayedCustomers');
+      cy.visit('/');
+      cy.wait('@getDelayedCustomers');
+      cy.contains('p', 'Loading...').should('be.visible');
+    });
+
+    context('Empty State', () => {
+      beforeEach(() => {
+        cy.intercept('GET', `${CUSTOMERS_URL}?page=1&limit=10&size=All&industry=All`, {
+          fixture: 'customers/emptyCustomers',
+        }).as('getEmptyCustomers');
+        cy.visit('/');
+        cy.wait('@getEmptyCustomers');
+      });
+
+      it('shows the image of an empty box and the text "No customers available." when there are no customers in the database', () => {
+        cy.get('svg[title="image of an empty box"]').should('be.visible');
+        cy.contains('span', 'No customers available.').should('be.visible');
+      });
+
+      it('disables the name text input field when there are no customers in the database', () => {
+        cy.get('#name').should('be.disabled');
+      });
+    });
+  });
+
+  context('Messenger Form', () => {
+    it('shows and hides a success message when successfully submitting the messenger form', () => {
+      const messengerData = {
+        name: 'Matheus',
+        email: 'matheus@gmail.com',
+        message: 'Hello, I need help with something.',
+      };
+
+      cy.getByClassStartsWith('Messenger_openCloseButton').click();
+      cy.get('#messenger-name').type(messengerData.name);
+      cy.get('#email').type(messengerData.email);
+      cy.get('#message').type(messengerData.message);
+      cy.clock();
+      cy.getByClassStartsWith('Messenger_sendButton').click();
+
+      cy.getByClassStartsWith('Messenger_success')
+        .should('be.visible')
+        .and('have.text', 'Your message has been sent.');
+      cy.tick(3000);
+      cy.getByClassStartsWith('Messenger_success').should('not.exist');
+    });
+  });
+
+  context('Filters', () => {
+    it('keeps the filters when coming back from the customer details view', () => {
+      const clientFilterData = {
+        size: 'Small',
+        industry: 'Technology',
+      };
+
+      cy.intercept(
+        'GET',
+        `${CUSTOMERS_URL}?page=1&limit=10&size=${clientFilterData.size}&industry=${clientFilterData.industry}`,
+        { fixture: 'customers/smallTechnologyCustomers' }
+      ).as('getSmallTechnologyCustomers');
+      cy.findByTestId('size-filter').select(clientFilterData.size);
+      cy.findByTestId('industry-filter').select(clientFilterData.industry);
+      cy.contains('button', 'View').click();
```

Comment:
```suggestion
      cy.contains('button', 'View').click();

```

Até aqui são pré-condições.

Clicar no botão _Back_ é a ação.

E o resto são as _assertions_.